broker/src/main/java/io/zeebe/broker/system/management/BrokerAdminServiceImpl.java

```diff
@@ -83,16 +96,23 @@ public void prepareForUpgrade() {
 
   private CompletableFuture<PartitionStatus> getPartitionStatus(final ZeebePartition partition) {
     final CompletableFuture<PartitionStatus> partitionStatus = new CompletableFuture<>();
-    getStreamProcessor(partition)
-        .onComplete(
-            (streamProcessor, throwable) -> {
-              if (throwable != null) {
-                partitionStatus.completeExceptionally(throwable);
+    final var streamProcessorFuture = toCompletableFuture(partition.getStreamProcessor());
+    final var exporterDirectorFuture = toCompletableFuture(partition.getExporterDirector());
+    CompletableFuture.allOf(streamProcessorFuture, exporterDirectorFuture)
+        .whenComplete(
+            (nothing, error) -> {
+              if (error != null) {
+                partitionStatus.completeExceptionally(error);
                 return;
               }
-              streamProcessor.ifPresentOrElse(
-                  sp -> getLeaderPartitionStatus(partition, sp, partitionStatus),
-                  () -> getFollowerPartitionStatus(partition, partitionStatus));
+              final var streamProcessor = streamProcessorFuture.join();
+              final var exporterDirector = exporterDirectorFuture.join();
+              if (streamProcessor.isPresent() && exporterDirector.isPresent()) {
```

Comment:
So we have a optional in a future? sounds like duplicated complexity :thinking:

Reply:
It's optional because followers doesn't have streamProcessor.

Reply:
but then it can be just null not ? :sweat_smile: Idk just have the feeling it is unnecessary

Reply:
It can be null. Then instead of 
`if(streamProcessor.isPresent() && exporterDirector.isPresent())` 
We would write `if(streamProcessor != null && exporterDirector != null)` . How is that better? Also I thought we want to use `Optional` in such cases.

Reply:
Yeah to use methods like `.orElse` or `ifPresentThen` etc but just to check whether is present I see no point in creating two separate objects for it. Not a fan of it.

Reply:
You can keep it but it just wanted to mention this.